test/common/network/io_socket_handle_impl_test.cc

```diff
@@ -79,25 +82,34 @@ TEST(IoSocketHandleImpl, LastRoundTripTimeReturnsRttIfSuccessful) {
   auto os_calls =
       std::make_unique<Envoy::TestThreadsafeSingletonInjector<Envoy::Api::OsSysCallsImpl>>(
           &os_sys_calls);
-  EXPECT_CALL(os_sys_calls, getsockopt_(_, _, _, _, _))
-      .WillOnce(DoAll(WithArg<3>(Invoke([](void* optval) {
-                        static_cast<struct tcp_info*>(optval)->tcpi_rtt = 35;
-                      })),
-                      Return(0)));
+
+  EXPECT_CALL(os_sys_calls, socketTcpInfo(_, _))
+      .WillOnce(Invoke([this](os_fd_t sockfd, tcp_info* tcpInfo) -> Api::SysCallBoolResult {
+        tcpInfo->tcpi_rtt = 35;
+        return {true, 0};
+      }));
 
   IoSocketHandleImpl io_handle;
   EXPECT_THAT(io_handle.lastRoundTripTime(), Eq(absl::optional<std::chrono::milliseconds>{35}));
 }
 
-#endif
-
-#ifndef TCP_INFO
-
-TEST(IoSocketHandleImpl, LastRoundTripTimeAlwaysReturnsEmptyOptional) {
-  IoSocketHandleImpl io_handle;
-  EXPECT_THAT(io_handle.lastRoundTripTime(), Eq(absl::optional<std::chrono::milliseconds>{}));
+// Only do the integration tests in supported platforms.
+#if defined(TCP_INFO) || defined(SIO_TCP_INFO)
+TEST(IoSocketHandleImpl, LastRoundTripIntegrationTest) {
+  struct sockaddr_in server;
+  // TCP info can not be calculated on loopback.
```

Comment:
Is this true on both platforms?

Reply:
I got this based on a comment on slack for linux, haven't personally tried it. On Windows that I tried it did not work with not supported error.